gcc/rust/typecheck/rust-unify.cc

```diff
@@ -122,7 +122,7 @@ UnifyRules::emit_type_mismatch () const
   rich_location r (line_table, locus);
   r.add_range (lhs.get_locus ());
   r.add_range (rhs.get_locus ());
-  rust_error_at (r, "expected %<%s%> got %<%s%>",
+  rust_error_at (r, ErrorCode::E0308, "mismatched types, expected %qs got %qs",
```

Comment:
```suggestion
  rust_error_at (r, ErrorCode::E0308, "mismatched types, expected %qs, got %qs",
```
I think we need a separator here, so either a comma or
```suggestion
  rust_error_at (r, ErrorCode::E0308, "mismatched types, expected %qs but got %qs",
```

Reply:
This one is better,
Sorry for these grammatical errors, I'll take care of these next time :D